app/Helpers/database/achievement.php

```diff
@@ -272,6 +273,8 @@ function UploadNewAchievement(
         $achievement->save();
         $idInOut = $achievement->ID;
 
+        $achievement->upsertAuthorshipCredit($author, AchievementAuthorTask::LOGIC);
```

Comment:
When an achievement is uploaded to the server, we immediately credit the uploader with logic credit.

Under the current design, everything else (writing, art, etc) is assumed unless someone or something adds credit for those tasks.